src/testModel.ts

```diff
@@ -715,11 +713,13 @@ export class TestModelCollection extends DisposableBase {
     if (model.isEnabled === enabled)
       return;
     model.isEnabled = enabled;
-    if (userGesture)
-      this._saveSettings();
     model.reset();
     const configSettings = this._configSettings(model.config);
-    model._loadModelIfNeeded(configSettings).then(() => this._didUpdate.fire());
+    model._loadModelIfNeeded(configSettings).then(() => {
+      this._didUpdate.fire();
+      if (userGesture)
+        this._saveSettings();
```

Comment:
Before, we were saving settings immediately before reading them in `this._configSettings`. This breaks the "no settings yet" path in `_loadModelIfNeeded`, and broke tests in`track-config.spec.ts`. By moving the save settings call after the `_loadModel`, we keep the "no settings yet" path working.

Reply:
How is the following case handled:

- I am like 90% of our users
- I created a playwright project a year ago, I only have one config, 'chromium' was checked for me automatically
- For the last year I was pressing "Run" button and I was getting all my tests run against Chromium
- I don't expect this to change when I update extension.

Reply:
Well when I wrote this I tried to change it a way that existing users don't have any changes. This is a complicated thing though, so let me try writing a test to verify this.

Reply:
Oh no :/ If the user never made a gesture that invokes `_saveSettings`, then their workspace settings are undefined, making them indistinguishable from users opening a new workspace. Not sure why, but I somehow assumed that we'd save settings immediately on opening. I'll have to rethink this.